
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />
<script type="text/javascript">

      var _gaq = _gaq || [];
      _gaq.push(['_setAccount', 'UA-139058009-1']);
      _gaq.push(['_trackPageview']);

      (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
      })();
    </script>
    <title>Commands from section ‘annotation’ &#8212; gtftk 1.6.2 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/nature.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Commands from section ‘ologram’" href="ologram.html" />
    <link rel="prev" title="Commands from section ‘conversion’" href="conversion.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="ologram.html" title="Commands from section ‘ologram’"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="conversion.html" title="Commands from section ‘conversion’"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">gtftk 1.6.2 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Commands from section ‘annotation’</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="commands-from-section-annotation">
<h1>Commands from section ‘annotation’<a class="headerlink" href="#commands-from-section-annotation" title="Permalink to this headline">¶</a></h1>
<p>In the example of this section we will need the following example files:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk get_example -q -d simple -f &#39;*&#39;
</pre></div>
</div>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk get_example -q -d mini_real -f &#39;*&#39;
</pre></div>
</div>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk get_example -q -d hg38_chr1 -f &#39;*&#39;
</pre></div>
</div>
<hr class="docutils" />
<section id="closest-genes">
<h2>closest_genes<a class="headerlink" href="#closest-genes" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Find the n closest genes for each transcript.</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk closest_genes  -i simple.gtf -f
genes	closest_genes	distances
G0001	G0007	18
G0002	G0010	4
G0003	G0004	4
G0004	G0003	4
G0005	G0006	12
G0006	G0005	12
G0007	G0001	18
G0008	G0002	42
G0009	G0006	21
G0010	G0002	4
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk closest_genes -h
  Usage: gtftk closest_genes [-i GTF] [-o GTF/TXT] [-r {tss,tts,gene}] [-nb nb_neighbors] [-t {tss,tts,gene}] [-s] [-S] [-f] [-H] [-k] [-id {gene_id,gene_name}] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Find the n closest genes for each genes.

  Notes:
     *  The reference region for each gene can be the TSS (the most 5&#39;), the TTS (The most 3&#39;) or
     the whole gene.
     *  The reference region for each closest gene can be the TSS, the whole gene or the TTS.
     *  The closest genes can be searched in a stranded or unstranded fashion.

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)

Arguments:
 -r, --from-region-type       What is region to consider for each gene. (default: tss)
 -nb, --nb-neighbors          The size of the neighborhood. (default: 1)
 -t, --to-region-type         What is region to consider for each closest gene. (default: tss)
 -s, --same-strandedness      Require same strandedness (default: False)
 -S, --diff-strandedness      Require different strandedness (default: False)
 -f, --text-format            Return a text format. (default: False)
 -H, --no-header              Don&#39;t print the header line. (default: False)
 -k, --collapse               Unwrap. Don&#39;t use comma. Print closest genes line by line. (default: False)
 -id, --identifier            The key used as gene identifier. (default: gene_id)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="overlapping">
<h2>overlapping<a class="headerlink" href="#overlapping" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Find transcripts whose body/TSS/TTS region extended in 5’ and 3’ (-u/-d) overlaps with any transcript from another gene. Strandness is not considered by default. Used –invert-match to find those that do not overlap. If –annotate-gtf is used, all lines of the input GTF file will be printed and a new key containing the list of overlapping transcripts will be added to the transcript features/lines (key will be ‘overlapping_*’ with * one of body/TSS/TTS). The –annotate-gtf and –invert-match arguments are mutually exclusive.</p>
<p><strong>Example:</strong> Find transcript whose promoter overlap transcript from other genes.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk overlapping -i simple.gtf -c simple.chromInfo -t promoter -u 10 -d 10 -a    | gtftk select_by_key -k feature -v transcript | gtftk tabulate -k transcript_id,overlap_promoter_u0.01k_d0.01k | head
transcript_id	overlap_promoter_u0.01k_d0.01k
G0001T002	G0007T001,G0007T002
G0001T001	G0007T001,G0007T002
G0002T001	G0010T001
G0003T001	G0004T002,G0004T001
G0004T002	G0003T001
G0004T001	G0003T001
G0005T001	G0003T001
G0006T001	G0005T001
G0006T002	G0005T001
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk overlapping -h
  Usage: gtftk overlapping [-i GTF] [-o GTF] -c CHROMINFO [-u UPSTREAM] [-d DOWNSTREAM] [-t {transcript,promoter,tts}] [-s] [-S] [-n] [-a] [-k key_name] [-b] [-@] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Find transcripts whose body/TSS/TTS region extended in 5&#39; and 3&#39; (-u/-d) overlaps with any
     transcript from another gene. Strandness is not considered by default. Used --invert-match to
     find those that do not overlap. If --annotate-gtf is used, all lines of the input GTF file will
     be printed and a new key containing the list of overlapping transcripts will be added to the
     transcript features/lines (key will be &#39;overlapping_*&#39; with * one of body/TSS/TTS). The
     --annotate-gtf and --invert-match arguments are mutually exclusive.

  Notes:
     *  --chrom-info may also accept &#39;mm8&#39;, &#39;mm9&#39;, &#39;mm10&#39;, &#39;hg19&#39;, &#39;hg38&#39;, &#39;rn3&#39; or &#39;rn4&#39;. In this
     case the  corresponding size of conventional chromosomes are used. ChrM is not used.

Arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -c, --chrom-info             Chromosome information. A tabulated two-columns file with chromosomes as column 1 and sizes as column 2 (default: None)
 -u, --upstream               Extend the region in 5&#39;. Used to define the region around the TSS/TTS. (default: 1500)
 -d, --downstream             Extend the region in 3&#39;. Used to define the region around the TSS/TTS. (default: 1500)
 -t, --feature-type           The feature of interest. (default: transcript)
 -s, --same-strandedness      Require same strandedness (default: False)
 -S, --diff-strandedness      Require different strandedness (default: False)
 -n, --invert-match           Not/Invert match. (default: False)
 -a, --annotate-gtf           All lines of the original GTF will be printed. (default: False)
 -k, --key-name               The name of the key. (default: None)
 -b, --bool                   When --annotate-gtf is used use 0 or 1 as key values (instead of overlapping transcripts id). (default: False)
 -@, --annotate-all           When --annotate-gtf annotate all transcripts (default value would be &#39;0&#39;). (default: False)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="divergent">
<h2>divergent<a class="headerlink" href="#divergent" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Find transcript with divergent promoters. These transcripts will be defined here
as those whose promoter region (defined by -u/-d) overlaps with the tss of
another gene in reverse/antisens orientation. This may be useful to select
coding genes in head-to-head orientation or LUAT as described in “Divergent
transcription is associated with promoters of transcriptional regulators”
(Lepoivre C, BMC Genomics, 2013). The output is a GTF with an additional key
(‘divergent’) whose value is set to ‘.’ if the gene has no antisens transcript
in its promoter region. If the gene has an antisens transcript in its promoter
region the ‘divergent’ key is set to the identifier of the transcript whose tss
is the closest relative to the considered promoter. The tss to tss distance is
also provided as an additional key (dist_to_divergent).</p>
<p><strong>Example:</strong> Flag divergent transcripts in the example dataset. Select them and produce a tabulated output.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk divergent -i simple.gtf -c simple.chromInfo -u 10 -d 10| gtftk select_by_key -k feature -v transcript | gtftk tabulate -k transcript_id,divergent,dist_to_divergent | head  -n 7
transcript_id	divergent	dist_to_divergent
G0003T001	G0004T002	4
G0004T002	G0003T001	4
G0004T001	G0003T001	4
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk divergent -h
  Usage: gtftk divergent [-i GTF] [-o GTF] -c CHROMINFO [-u UPSTREAM] [-d DOWNSTREAM] [-n] [-S] [-a key_name] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Find transcripts with divergent promoters. These transcripts will be defined here as those whose
     promoter region (defined by -u/-d) overlaps with the tss of another gene in reverse/antisens
     orientation. This may be useful to select coding genes in head-to-head orientation or LUAT as
     described in &quot;Divergent transcription is associated with promoters of transcriptional
     regulators&quot; (Lepoivre C, BMC Genomics, 2013). The output is a GTF with an additional key
     (&#39;divergent&#39;) whose value is set to &#39;.&#39; if the gene has no antisens transcript in its promoter
     region. If the gene has an antisens transcript in its promoter region the &#39;divergent&#39; key is
     set to the identifier of the transcript whose tss is the closest relative to the considered
     promoter. The tss to tss distance is also provided as an additional key (dist_to_divergent).

  Notes:
     *  chrom-info may also accept &#39;mm8&#39;, &#39;mm9&#39;, &#39;mm10&#39;, &#39;hg19&#39;, &#39;hg38&#39;, &#39;rn3&#39; or &#39;rn4&#39;. In this
     case the  corresponding size of conventional chromosomes are used. To get the size of  the
     chromosome in ensembl format (whithout chr prefix), use &#39;mm8_ens&#39;, &#39;mm9_ens&#39;,  &#39;mm10_ens&#39;,
     &#39;hg19_ens&#39;, &#39;hg38_ens&#39;, &#39;rn3_ens&#39; or &#39;rn4_ens&#39;. ChrM is not used.

Arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -c, --chrom-info             Tabulated two-columns file. Chromosomes as column 1 and their sizes as column 2 (default: None)
 -u, --upstream               Extend the promoter in 5&#39; by a given value (int). Defines the region around the tss. (default: 1500)
 -d, --downstream             Extend the region in 3&#39; by a given value (int). Defines the region around the tss. (default: 1500)
 -n, --no-annotation          Do not annotate the GTF. Just select the divergent transcripts. (default: False)
 -S, --no-strandness          Do not consider strandness (only look whether the promoter from a transcript overlaps with the promoter from another gene). (default: False)
 -a, --key-name               The name of the key. (default: None)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="convergent">
<h2>convergent<a class="headerlink" href="#convergent" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Find transcript with convergent tts. These transcripts will be defined here
as those whose tts region (defined by -u/-d) overlaps with the tts of
another gene in reverse/antisens orientation. The output is a GTF with an
additional key (‘convergent’) whose value is set to ‘.’ if the gene has no
convergent transcript in its tts region. If the gene has an antisens transcript
in its tts region the ‘convergent’ key is set to the identifier of the
transcript whose tts is the closest relative to the considered tts.
The tts to tts distance is also provided as an additional key (dist_to_convergent).</p>
<p><strong>Example:</strong> Flag divergent transcripts in the example dataset. Select them and produce a tabulated output.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk convergent -i simple.gtf -c simple.chromInfo -u 25 -d 25| gtftk select_by_key -k feature -v transcript | gtftk tabulate -k transcript_id,convergent,dist_to_convergent| head -n 4
transcript_id	convergent	dist_to_convergent
G0002T001	G0008T001	21
G0008T001	G0002T001	21
G0010T001	G0008T001	24
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk convergent -h
  Usage: gtftk convergent [-i GTF] [-o GTF] -c CHROMINFO [-u UPSTREAM] [-d DOWNSTREAM] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Find transcripts with convergent tts. These transcripts will be defined here as those whose tts
     region (defined by -u/-d) overlaps with the tts of another gene in reverse/antisens
     orientation. The output is a GTF with an additional key (&#39;convergent&#39;) whose value is set to
     &#39;.&#39; if the gene has no convergent transcript in its tts region. If the gene has an antisens
     transcript in its tts region the &#39;convergent&#39; key is set to the identifier of the transcript
     whose tts is the closest relative to the considered tts. The tts to tts distance is also
     provided as an additional key (dist_to_convergent).

  Notes:
     *  chrom-info may also accept &#39;mm8&#39;, &#39;mm9&#39;, &#39;mm10&#39;, &#39;hg19&#39;, &#39;hg38&#39;, &#39;rn3&#39; or &#39;rn4&#39;. In this
     case the  corresponding size of conventional chromosomes are used. To get the size of  the
     chromosome in ensembl format (whithout chr prefix), use &#39;mm8_ens&#39;, &#39;mm9_ens&#39;,  &#39;mm10_ens&#39;,
     &#39;hg19_ens&#39;, &#39;hg38_ens&#39;, &#39;rn3_ens&#39; or &#39;rn4_ens&#39;. ChrM is not used.

Arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -c, --chrom-info             Tabulated two-columns file. Chromosomes as column 1 and sizes as column 2 (default: None)
 -u, --upstream               Extends the tts in 5&#39; by a given value (int). Defines the region around the tts. (default: 1500)
 -d, --downstream             Extends the region in 3&#39; by a given value (int). Defines the region around the tts. (default: 1500)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="exon-sizes">
<h2>exon_sizes<a class="headerlink" href="#exon-sizes" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Add a new key to transcript features containing a comma-separated list of exon sizes.</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk exon_sizes -i simple.gtf | gtftk select_by_key -t | gtftk tabulate -k transcript_id,exon_sizes
transcript_id	exon_sizes
G0001T002	14
G0001T001	14
G0002T001	10
G0003T001	5,5
G0004T002	4,1,3
G0004T001	4,1,3
G0005T001	6,3
G0006T001	3,3,4
G0006T002	3,3
G0007T001	10
G0007T002	10
G0008T001	3,5
G0009T002	12
G0009T001	12
G0010T001	11
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk exon_sizes -h
  Usage: gtftk exon_sizes [-i GTF] [-o TXT] [-a key_name] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Add a new key to transcript features containing a comma-separated list of exon sizes.

Arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN. (default: &lt;stdin&gt;)
 -o, --outputfile             Output GTF file. (default: &lt;stdout&gt;)
 -a, --key-name               The name of the key. (default: exon_sizes)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="intron-sizes">
<h2>intron_sizes<a class="headerlink" href="#intron-sizes" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Add a new key to transcript features containing a comma-separated list of intron sizes.</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk intron_sizes -i simple.gtf | gtftk select_by_key -t | gtftk tabulate -k transcript_id,intron_sizes
transcript_id	intron_sizes
G0001T002	0
G0001T001	0
G0002T001	0
G0003T001	2
G0004T002	2,2
G0004T001	2,2
G0005T001	6
G0006T001	2,2
G0006T002	2
G0007T001	0
G0007T002	0
G0008T001	5
G0009T002	0
G0009T001	0
G0010T001	0
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk intron_sizes -h
  Usage: gtftk intron_sizes [-i GTF] [-o GTF] [-a key_name] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Add a new key to transcript features containing a comma-separated list of intron-size.

Arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN. (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -a, --key-name               The name of the key. (default: intron_sizes)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">Commands from section ‘annotation’</a><ul>
<li><a class="reference internal" href="#closest-genes">closest_genes</a></li>
<li><a class="reference internal" href="#overlapping">overlapping</a></li>
<li><a class="reference internal" href="#divergent">divergent</a></li>
<li><a class="reference internal" href="#convergent">convergent</a></li>
<li><a class="reference internal" href="#exon-sizes">exon_sizes</a></li>
<li><a class="reference internal" href="#intron-sizes">intron_sizes</a></li>
</ul>
</li>
</ul>

  </div>
  <div>
    <h4>Previous topic</h4>
    <p class="topless"><a href="conversion.html"
                          title="previous chapter">Commands from section ‘conversion’</a></p>
  </div>
  <div>
    <h4>Next topic</h4>
    <p class="topless"><a href="ologram.html"
                          title="next chapter">Commands from section ‘ologram’</a></p>
  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/annotation.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="ologram.html" title="Commands from section ‘ologram’"
             >next</a> |</li>
        <li class="right" >
          <a href="conversion.html" title="Commands from section ‘conversion’"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">gtftk 1.6.2 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Commands from section ‘annotation’</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, F. Lopez and D. Puthier.
      Last updated on Mar 24, 2022.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.4.0.
    </div>
  </body>
</html>