
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />
<script type="text/javascript">

      var _gaq = _gaq || [];
      _gaq.push(['_setAccount', 'UA-139058009-1']);
      _gaq.push(['_trackPageview']);

      (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
      })();
    </script>
    <title>Commands from section ‘miscellaneous’ &#8212; gtftk 1.6.2 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/nature.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Help on Python package (pygtftk)" href="api.html" />
    <link rel="prev" title="Commands from section ‘coverage’" href="coverage.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="api.html" title="Help on Python package (pygtftk)"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="coverage.html" title="Commands from section ‘coverage’"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">gtftk 1.6.2 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Commands from section ‘miscellaneous’</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="commands-from-section-miscellaneous">
<h1>Commands from section ‘miscellaneous’<a class="headerlink" href="#commands-from-section-miscellaneous" title="Permalink to this headline">¶</a></h1>
<p>In this section we will require the following datasets:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk get_example -q -d mini_real -f &#39;*&#39;
</pre></div>
</div>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk get_example -q -d simple -f &#39;*&#39;
</pre></div>
</div>
<section id="control-list">
<h2>control_list<a class="headerlink" href="#control-list" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Returns a list of gene matched for expression based on reference values. Based on a reference gene list (or more generally IDs) this command tries to extract a set of other genes/IDs matched for signal/expression. The –reference-gene-file contains the list of reference IDs while the –inputfile contains a tuple gene/signal for all genes.</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk control_list -i mini_real_counts_ENCFF630HEX.tsv -r mini_real_control_1.txt -D -V 2 -s -l -p 1 -ju -if example_13.png -pf png
 |-- 10:52-INFO-control_list : 0 duplicate lines have been deleted in reference file.
 |-- 10:52-INFO-control_list : Found 50 genes of the reference in the provided signal file
 |-- 10:52-INFO-control_list : All reference genes were found.
 |-- 10:52-INFO-control_list : Searching for genes with matched signal.
 |-- 10:52-INFO-control_list : Preparing a dataframe for plotting.
 |-- 10:52-INFO-control_list : Saving diagram to file : example_13.png
 |-- 10:52-INFO-control_list : Be patient. This may be long for large datasets.
</pre></div>
</div>
<a class="reference external image-reference" href="_static/example_13.png"><img alt="_images/example_13.png" src="_images/example_13.png" /></a>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk control_list -h
  Usage: gtftk control_list --in-file TXT --reference-gene-file TXT [--out-dir DIR] [--log2] [--pseudo-count pseudo_count] [-pw page_width] [-ph page_height] [-pf {pdf,png}] [-dpi dpi] [--skip-first] [--rug] [--jitter] [-if user_img_file] [-c set_colors] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Based on a reference gene list (or more generally IDs) this command tries to extract a set of
     other genes/IDs matched for signal/expression. The --reference-gene-file contains the list of
     reference IDs while the -\inputfile contains a tuple gene/signal for all genes.

  Notes:
     *  --infile is a two columns tabulated file. The first column contains the list of ids
     (including reference IDs) and the second column contains the expression/signal values. This
     file should contain no header.
     *  Think about discarding any unwanted IDs from --infile before calling control_list.

optional arguments:
 --in-file, -i                A two columns tab-file. See notes. (default: None)
 --reference-gene-file, -r    The file containing the reference gene list (1 column, transcript ids). No header. (default: None)
 --out-dir, -o                Name of the output directory. (default: control_list)
 --log2, -l                   If selected, data will be log transformed. (default: False)
 --pseudo-count, -p           The value for a pseudo-count to be added. (default: 0)
 -pw, --page-width            Output pdf file width (e.g. 7 inches). (default: None)
 -ph, --page-height           Output file height (e.g. 5 inches). (default: None)
 -pf, --page-format           Output file format. (default: pdf)
 -dpi, --dpi                  Dpi to use. (default: 300)
 --skip-first, -s             Indicates that infile hase a header. (default: False)
 --rug, -u                    Add rugs to the diagram. (default: False)
 --jitter, -j                 Add jittered points. (default: False)
 -if, --user-img-file         Provide an alternative path for the image. (default: None)
 -c, --set-colors             Colors for the two sets (comma-separated). (default: #b2df8a,#6a3d9a)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<section id="col-from-tab">
<h2>col_from_tab<a class="headerlink" href="#col-from-tab" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select columns from a tabulated file based on their names.</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_key -t -i simple.gtf | gtftk tabulate -k &#39;*&#39; -x | gtftk col_from_tab -c transcript_id,gene_id
transcript_id	gene_id
G0001T002	G0001
G0001T001	G0001
G0002T001	G0002
G0003T001	G0003
G0004T002	G0004
G0004T001	G0004
G0005T001	G0005
G0006T001	G0006
G0006T002	G0006
G0007T001	G0007
G0007T002	G0007
G0008T001	G0008
G0009T002	G0009
G0009T001	G0009
G0010T001	G0010
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk col_from_tab -h
  Usage: gtftk col_from_tab [-i TXT] [-o TXT] -c columns [-n] [-u] [-s SEP] [-r OUT_SEP] [-m MORE_COL] [-H] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select columns from a tabulated file based on their names.

Arguments:
 -i, --inputfile              The tabulated file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -c, --columns                The list (csv) of column names. (default: None)
 -n, --invert-match           Not/invert match. (default: False)
 -u, --unique                 Write non redondant lines. (default: False)
 -s, --separator              The separator of input columns. (default: )
 -r, --output-separator       The separator to be used for separating output columns. (default: )
 -m, --more-col               Add a named (last) column with a given value (e.g. -m col_name:value). (default: None)
 -H, --no-header              Don&#39;t print the header line. (default: False)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">Commands from section ‘miscellaneous’</a><ul>
<li><a class="reference internal" href="#control-list">control_list</a></li>
<li><a class="reference internal" href="#col-from-tab">col_from_tab</a></li>
</ul>
</li>
</ul>

  </div>
  <div>
    <h4>Previous topic</h4>
    <p class="topless"><a href="coverage.html"
                          title="previous chapter">Commands from section ‘coverage’</a></p>
  </div>
  <div>
    <h4>Next topic</h4>
    <p class="topless"><a href="api.html"
                          title="next chapter">Help on Python package (pygtftk)</a></p>
  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/miscellaneous.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="api.html" title="Help on Python package (pygtftk)"
             >next</a> |</li>
        <li class="right" >
          <a href="coverage.html" title="Commands from section ‘coverage’"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">gtftk 1.6.2 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Commands from section ‘miscellaneous’</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, F. Lopez and D. Puthier.
      Last updated on Mar 24, 2022.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.4.0.
    </div>
  </body>
</html>