
<!DOCTYPE html>

<html lang="en">
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="generator" content="Docutils 0.17.1: http://docutils.sourceforge.net/" />
<script type="text/javascript">

      var _gaq = _gaq || [];
      _gaq.push(['_setAccount', 'UA-139058009-1']);
      _gaq.push(['_trackPageview']);

      (function() {
        var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
        ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
        var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
      })();
    </script>
    <title>Commands from section ‘selection’ &#8212; gtftk 1.6.2 documentation</title>
    <link rel="stylesheet" type="text/css" href="_static/pygments.css" />
    <link rel="stylesheet" type="text/css" href="_static/nature.css" />
    <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
    <script src="_static/jquery.js"></script>
    <script src="_static/underscore.js"></script>
    <script src="_static/doctools.js"></script>
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Commands from section ‘conversion’" href="conversion.html" />
    <link rel="prev" title="Commands from section ‘Editing’" href="editing.html" /> 
  </head><body>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             accesskey="I">index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="conversion.html" title="Commands from section ‘conversion’"
             accesskey="N">next</a> |</li>
        <li class="right" >
          <a href="editing.html" title="Commands from section ‘Editing’"
             accesskey="P">previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">gtftk 1.6.2 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Commands from section ‘selection’</a></li> 
      </ul>
    </div>  

    <div class="document">
      <div class="documentwrapper">
        <div class="bodywrapper">
          <div class="body" role="main">
            
  <section id="commands-from-section-selection">
<h1>Commands from section ‘selection’<a class="headerlink" href="#commands-from-section-selection" title="Permalink to this headline">¶</a></h1>
<p>In this section we will require the following datasets:</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk get_example -q -d mini_real -f &#39;*&#39;
</pre></div>
</div>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk get_example -q -d tiny_real -f &#39;*&#39;
</pre></div>
</div>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk get_example -q -d simple -f &#39;*&#39;
</pre></div>
</div>
<section id="select-by-key">
<h2>select_by_key<a class="headerlink" href="#select-by-key" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Extract lines from the gtf based on key and values.</p>
<p><strong>Example:</strong> Select some gene_id.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_key -i simple.gtf -k gene_id -v G0002,G0003,G0004
chr1	gtftk	gene	180	189	.	+	.	gene_id &quot;G0002&quot;;
chr1	gtftk	transcript	180	189	.	+	.	gene_id &quot;G0002&quot;; transcript_id &quot;G0002T001&quot;;
chr1	gtftk	exon	180	189	.	+	.	gene_id &quot;G0002&quot;; transcript_id &quot;G0002T001&quot;; exon_id &quot;G0002T001E001&quot;;
chr1	gtftk	CDS	180	182	.	+	.	gene_id &quot;G0002&quot;; transcript_id &quot;G0002T001&quot;; ccds_id &quot;CDS_G0002T001&quot;;
chr1	gtftk	gene	50	61	.	-	.	gene_id &quot;G0003&quot;;
chr1	gtftk	transcript	50	61	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;;
chr1	gtftk	exon	50	54	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;; exon_id &quot;G0003T001E001&quot;;
chr1	gtftk	exon	57	61	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;; exon_id &quot;G0003T001E002&quot;;
chr1	gtftk	CDS	50	52	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;; ccds_id &quot;CDS_G0003T001&quot;;
chr1	gtftk	gene	65	76	.	+	.	gene_id &quot;G0004&quot;;
chr1	gtftk	transcript	65	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;;
chr1	gtftk	exon	65	68	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; exon_id &quot;G0004T002E001&quot;;
chr1	gtftk	exon	71	71	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; exon_id &quot;G0004T002E002&quot;;
chr1	gtftk	exon	74	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; exon_id &quot;G0004T002E003&quot;;
chr1	gtftk	CDS	66	68	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; ccds_id &quot;CDS_G0004T002&quot;;
chr1	gtftk	CDS	71	71	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; ccds_id &quot;CDS_G0004T002&quot;;
chr1	gtftk	CDS	74	75	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; ccds_id &quot;CDS_G0004T002&quot;;
chr1	gtftk	transcript	65	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;;
chr1	gtftk	exon	65	68	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;; exon_id &quot;G0004T001E001&quot;;
chr1	gtftk	exon	71	71	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;; exon_id &quot;G0004T001E002&quot;;
chr1	gtftk	exon	74	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;; exon_id &quot;G0004T001E003&quot;;
chr1	gtftk	CDS	65	67	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;; ccds_id &quot;CDS_G0004T001&quot;;
</pre></div>
</div>
<p><strong>Example:</strong> Select using basic attributes (chrom, source, feature…). Note that seqid, seqname and chrom are synonymous.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_key -i simple.gtf -k feature -v transcript,exon | gtftk select_by_key -k seqname -v chr1
chr1	gtftk	transcript	125	138	.	+	.	gene_id &quot;G0001&quot;; transcript_id &quot;G0001T002&quot;;
chr1	gtftk	exon	125	138	.	+	.	gene_id &quot;G0001&quot;; transcript_id &quot;G0001T002&quot;; exon_id &quot;G0001T002E001&quot;;
chr1	gtftk	transcript	125	138	.	+	.	gene_id &quot;G0001&quot;; transcript_id &quot;G0001T001&quot;;
chr1	gtftk	exon	125	138	.	+	.	gene_id &quot;G0001&quot;; transcript_id &quot;G0001T001&quot;; exon_id &quot;G0001T001E001&quot;;
chr1	gtftk	transcript	180	189	.	+	.	gene_id &quot;G0002&quot;; transcript_id &quot;G0002T001&quot;;
chr1	gtftk	exon	180	189	.	+	.	gene_id &quot;G0002&quot;; transcript_id &quot;G0002T001&quot;; exon_id &quot;G0002T001E001&quot;;
chr1	gtftk	transcript	50	61	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;;
chr1	gtftk	exon	50	54	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;; exon_id &quot;G0003T001E001&quot;;
chr1	gtftk	exon	57	61	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;; exon_id &quot;G0003T001E002&quot;;
chr1	gtftk	transcript	65	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;;
chr1	gtftk	exon	65	68	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; exon_id &quot;G0004T002E001&quot;;
chr1	gtftk	exon	71	71	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; exon_id &quot;G0004T002E002&quot;;
chr1	gtftk	exon	74	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; exon_id &quot;G0004T002E003&quot;;
chr1	gtftk	transcript	65	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;;
chr1	gtftk	exon	65	68	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;; exon_id &quot;G0004T001E001&quot;;
chr1	gtftk	exon	71	71	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;; exon_id &quot;G0004T001E002&quot;;
chr1	gtftk	exon	74	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;; exon_id &quot;G0004T001E003&quot;;
chr1	gtftk	transcript	33	47	.	-	.	gene_id &quot;G0005&quot;; transcript_id &quot;G0005T001&quot;;
chr1	gtftk	exon	33	35	.	-	.	gene_id &quot;G0005&quot;; transcript_id &quot;G0005T001&quot;; exon_id &quot;G0005T001E001&quot;;
chr1	gtftk	exon	42	47	.	-	.	gene_id &quot;G0005&quot;; transcript_id &quot;G0005T001&quot;; exon_id &quot;G0005T001E002&quot;;
chr1	gtftk	transcript	22	35	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T001&quot;;
chr1	gtftk	exon	22	25	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T001&quot;; exon_id &quot;G0006T001E001&quot;;
chr1	gtftk	exon	28	30	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T001&quot;; exon_id &quot;G0006T001E002&quot;;
chr1	gtftk	exon	33	35	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T001&quot;; exon_id &quot;G0006T001E003&quot;;
chr1	gtftk	transcript	28	35	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T002&quot;;
chr1	gtftk	exon	28	30	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T002&quot;; exon_id &quot;G0006T002E001&quot;;
chr1	gtftk	exon	33	35	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T002&quot;; exon_id &quot;G0006T002E002&quot;;
chr1	gtftk	transcript	107	116	.	+	.	gene_id &quot;G0007&quot;; transcript_id &quot;G0007T001&quot;;
chr1	gtftk	exon	107	116	.	+	.	gene_id &quot;G0007&quot;; transcript_id &quot;G0007T001&quot;; exon_id &quot;G0007T001E001&quot;;
chr1	gtftk	transcript	107	116	.	+	.	gene_id &quot;G0007&quot;; transcript_id &quot;G0007T002&quot;;
chr1	gtftk	exon	107	116	.	+	.	gene_id &quot;G0007&quot;; transcript_id &quot;G0007T002&quot;; exon_id &quot;G0007T002E001&quot;;
chr1	gtftk	transcript	210	222	.	-	.	gene_id &quot;G0008&quot;; transcript_id &quot;G0008T001&quot;;
chr1	gtftk	exon	210	214	.	-	.	gene_id &quot;G0008&quot;; transcript_id &quot;G0008T001&quot;; exon_id &quot;G0008T001E001&quot;;
chr1	gtftk	exon	220	222	.	-	.	gene_id &quot;G0008&quot;; transcript_id &quot;G0008T001&quot;; exon_id &quot;G0008T001E002&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;;
chr1	gtftk	exon	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;; exon_id &quot;G0009T002E001&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;;
chr1	gtftk	exon	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;; exon_id &quot;G0009T001E001&quot;;
chr1	gtftk	transcript	176	186	.	+	.	gene_id &quot;G0010&quot;; transcript_id &quot;G0010T001&quot;;
chr1	gtftk	exon	176	186	.	+	.	gene_id &quot;G0010&quot;; transcript_id &quot;G0010T001&quot;; exon_id &quot;G0010T001E001&quot;;
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_key -h
  Usage: gtftk select_by_key [-i GTF] [-o GTF] [-k KEY] [-v VALUE] [-f FILE] [-c COL] [-n] [-b] [-m NAME] [-s SEP] [-l] [-t] [-g] [-e] [-d] [-a] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select lines from a GTF file based on attributes and associated values.

optional arguments:
 -v, --value                  A comma-separated list of values. (default: None)
 -f, --file-with-values       A file containing values as a single column. (default: None)
 -t, --select-transcripts     A shortcuts for &quot;-k feature -v transcript&quot;. (default: False)
 -g, --select-genes           A shortcuts for &quot;-k feature -v gene&quot;. (default: False)
 -e, --select-exons           A shortcuts for &quot;-k feature -v exon&quot;. (default: False)
 -d, --select-cds             A shortcuts for &quot;-k feature -v CDS&quot;. (default: False)
 -a, --select-start-codon     A shortcuts for &quot;-k feature -v start_codon&quot;. (default: False)

Arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -k, --key                    The key name. (default: None)
 -c, --col                    The column number (one-based) that contains the values in the file. File is tab-delimited. (default: 1)
 -n, --invert-match           Not/invert match. Select lines whose selected key is not associated with the selected values. (default: False)
 -b, --bed-format             Ask for bed format output. (default: False)
 -m, --names                  If Bed output. The key(s) that should be used as name. (default: gene_id,transcript_id)
 -s, --separator              If Bed output. The separator to be used for separating name elements (see -n). (default: |)
 -l, --log                    Print some statistics about selected features. To be used in conjunction with -V 1/2. (default: False)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-by-regexp">
<h2>select_by_regexp<a class="headerlink" href="#select-by-regexp" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select lines by testing values of a particular key with a regular expression</p>
<p><strong>Example:</strong> Select lines corresponding to gene_names matching the regular expression ‘G.*9$’.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_regexp -i simple.gtf -k gene_id -r &#39;G.*9$&#39;
chr1	gtftk	gene	3	14	.	-	.	gene_id &quot;G0009&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;;
chr1	gtftk	exon	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;; exon_id &quot;G0009T002E001&quot;;
chr1	gtftk	CDS	5	10	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;; ccds_id &quot;CDS_G0009T002&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;;
chr1	gtftk	exon	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;; exon_id &quot;G0009T001E001&quot;;
chr1	gtftk	CDS	3	8	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;; ccds_id &quot;CDS_G0009T001&quot;;
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_regexp -h
  Usage: gtftk select_by_regexp [-i GTF] [-o GTF] [-k KEY] [-r regexp] [-n] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select lines from a GTF file based on a regexp.

  Notes:
     *  The default is to try to select feature from conventional human chromosome (chr1..chr22,
     chrX and chrY) with --key set to chrom and --regexp set to &quot;^chr[0-9XY]+$&quot;.

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -k, --key                    The key name (default: chrom)
 -r, --regexp                 The regular expression. (default: ^chr[0-9XY]+$)
 -n, --invert-match           Not/invert match. Selected lines whose requested key do not match the regexp. (default: False)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-by-intron-size">
<h2>select_by_intron_size<a class="headerlink" href="#select-by-intron-size" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Delete genes containing an intron whose size is below s. If -m is selected, any gene whose sum of intronic region length is above s is deleted. Monoexonic genes are kept.</p>
<p><strong>Example:</strong> Some genes having transcripts containing an intron whose size is below 80 nucleotides</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_intron_size -s 200 -vd -i tiny_real.gtf.gz | gtftk intron_sizes | gtftk tabulate -k gene_name,transcript_id,intron_sizes -Hun
MCAM	ENST00000526992	159,98
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_regexp -h
  Usage: gtftk select_by_regexp [-i GTF] [-o GTF] [-k KEY] [-r regexp] [-n] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select lines from a GTF file based on a regexp.

  Notes:
     *  The default is to try to select feature from conventional human chromosome (chr1..chr22,
     chrX and chrY) with --key set to chrom and --regexp set to &quot;^chr[0-9XY]+$&quot;.

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -k, --key                    The key name (default: chrom)
 -r, --regexp                 The regular expression. (default: ^chr[0-9XY]+$)
 -n, --invert-match           Not/invert match. Selected lines whose requested key do not match the regexp. (default: False)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-by-max-exon-nb">
<h2>select_by_max_exon_nb<a class="headerlink" href="#select-by-max-exon-nb" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> For each gene select the transcript with the highest number of exons.</p>
<p><strong>Example:</strong> Select lines corresponding to gene_names matching the regular expression ‘BCL.*’.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_max_exon_nb -i simple.gtf | gtftk select_by_key -t
chr1	gtftk	transcript	125	138	.	+	.	gene_id &quot;G0001&quot;; transcript_id &quot;G0001T002&quot;;
chr1	gtftk	transcript	180	189	.	+	.	gene_id &quot;G0002&quot;; transcript_id &quot;G0002T001&quot;;
chr1	gtftk	transcript	50	61	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;;
chr1	gtftk	transcript	65	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;;
chr1	gtftk	transcript	33	47	.	-	.	gene_id &quot;G0005&quot;; transcript_id &quot;G0005T001&quot;;
chr1	gtftk	transcript	22	35	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T001&quot;;
chr1	gtftk	transcript	107	116	.	+	.	gene_id &quot;G0007&quot;; transcript_id &quot;G0007T001&quot;;
chr1	gtftk	transcript	210	222	.	-	.	gene_id &quot;G0008&quot;; transcript_id &quot;G0008T001&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;;
chr1	gtftk	transcript	176	186	.	+	.	gene_id &quot;G0010&quot;; transcript_id &quot;G0010T001&quot;;
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_max_exon_nb -h
  Usage: gtftk select_by_max_exon_nb [-i GTF] [-o GTF] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     For each gene select the transcript with the highest number of exons. If ties, select the first
     encountered.

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-by-loc">
<h2>select_by_loc<a class="headerlink" href="#select-by-loc" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select transcripts/gene overlapping a given locations. A transcript is defined here as the genomic region from TSS to TTS including introns. This function will return the transcript and all its associated elements (exons, utr…) even if only a fraction (e.g intron) of the transcript is overlapping the feature. If -/-ft-type is set to ‘gene’ returns the gene and all its associated elements.</p>
<p><strong>Example:</strong> Select transcripts at a given location.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_key -k feature -v transcript -i simple.gtf | gtftk  select_by_loc -l chr1:10-15
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;;
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_loc -h
  Usage: gtftk select_by_loc [-i GTF] [-o GTF] (-l LOC | -f BEDFILE) [-t {transcript,gene}] [-n] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select transcripts/gene overlapping a given locations.

  Notes:
     *  A transcript is defined here as the genomic region from TSS to TTS including introns.
     *  This function will return the transcript and all its associated elements (exons, utr...)
     even if only a fraction (e.g intron) of the transcript is overlapping the feature.
     *  If -/-ft-type is set to &#39;gene&#39; returns the gene and all its associated elements.

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -l, --location               List of chromosomal locations (chr:start-end[,chr:start-end]). 0-based (default: None)
 -f, --location-file          Bed file with chromosomal location. (default: None)
 -t, --ft-type                The feature of interest. (default: transcript)
 -n, --invert-match           Not/invert match. Select transcript not overlapping. (default: False)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-by-nb-exon">
<h2>select_by_nb_exon<a class="headerlink" href="#select-by-nb-exon" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select transcripts based on the number of exons.</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_nb_exon -m 2 -i simple.gtf | gtftk nb_exons| gtftk select_by_key -t
chr1	gtftk	transcript	50	61	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;; nb_exons &quot;2&quot;;
chr1	gtftk	transcript	65	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T002&quot;; nb_exons &quot;3&quot;;
chr1	gtftk	transcript	65	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;; nb_exons &quot;3&quot;;
chr1	gtftk	transcript	33	47	.	-	.	gene_id &quot;G0005&quot;; transcript_id &quot;G0005T001&quot;; nb_exons &quot;2&quot;;
chr1	gtftk	transcript	22	35	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T001&quot;; nb_exons &quot;3&quot;;
chr1	gtftk	transcript	28	35	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T002&quot;; nb_exons &quot;2&quot;;
chr1	gtftk	transcript	210	222	.	-	.	gene_id &quot;G0008&quot;; transcript_id &quot;G0008T001&quot;; nb_exons &quot;2&quot;;
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_nb_exon -h
  Usage: gtftk select_by_nb_exon [-i GTF] [-o GTF] [-m min_exon_number] [-M max_exon_number] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select transcripts based on the number of exons.

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -m, --min-exon-number        Minimum number of exons. (default: 0)
 -M, --max-exon-number        Maximum number of exons. (default: None)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-by-numeric-value">
<h2>select_by_numeric_value<a class="headerlink" href="#select-by-numeric-value" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select lines from a GTF file based on a boolean test on numeric values.</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk join_attr -i simple.gtf  -j simple.join_mat -k gene_id -m|  gtftk select_by_numeric_value -t &#39;start &lt; 10 and end &gt; 10 and S1 == 0.5555 and S2 == 0.7&#39; -n &quot;.,?&quot;
chr1	gtftk	gene	3	14	.	-	.	gene_id &quot;G0009&quot;; S1 &quot;0.5555&quot;; S2 &quot;0.7&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;; S1 &quot;0.5555&quot;; S2 &quot;0.7&quot;;
chr1	gtftk	exon	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T002&quot;; exon_id &quot;G0009T002E001&quot;; S1 &quot;0.5555&quot;; S2 &quot;0.7&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;; S1 &quot;0.5555&quot;; S2 &quot;0.7&quot;;
chr1	gtftk	exon	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;; exon_id &quot;G0009T001E001&quot;; S1 &quot;0.5555&quot;; S2 &quot;0.7&quot;;
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_numeric_value -h
  Usage: gtftk select_by_numeric_value [-i GTF] [-o GTF] -t test [-n na_omit] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select lines from a GTF file based on a boolean test on numeric values.

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -t, --test                   The test to be applied. (default: None)
 -n, --na-omit                If one of the evaluated values is enclosed in this list (csv), line is skipped. (default: None)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="random-list">
<h2>random_list<a class="headerlink" href="#random-list" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select a random list of genes or transcripts.</p>
<p><strong>Example:</strong> Select randomly 3 transcripts.</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk random_list -n 3 -i simple.gtf | gtftk count
transcript	3
exon	6
CDS	5
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk random_list -h
  Usage: gtftk random_list [-i GTF] [-o GTF] [-n NUMBER] [-t {gene,transcript}] [-s SEED] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select a random list of genes or transcripts. Note that if transcripts are requested the &#39;gene&#39;
     feature is not returned.

Arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -n, --number                 The number of transcripts or gene to select. (default: 1)
 -t, --ft-type                The type of feature. (default: transcript)
 -s, --seed-value             Seed value for the random number generator. (default: None)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="random-tx">
<h2>random_tx<a class="headerlink" href="#random-tx" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select randomly up to m transcript for each gene.</p>
<p><strong>Example:</strong> Select randomly 1 transcript per gene (<em>-m 1</em>).</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk random_tx -m 1 -i simple.gtf | gtftk select_by_key -k feature -v gene,transcript| gtftk tabulate -k gene_id,transcript_id
gene_id	transcript_id
G0001	G0001T001
G0002	G0002T001
G0003	G0003T001
G0004	G0004T001
G0005	G0005T001
G0006	G0006T001
G0007	G0007T002
G0008	G0008T001
G0009	G0009T002
G0010	G0010T001
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk random_tx -h
  Usage: gtftk random_tx [-i GTF] [-o GTF] [-m MAX] [-s SEED] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select randomly up to m transcript for each gene.

Arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -m, --max-transcript         The maximum number of transcripts to select for each gene. (default: 1)
 -s, --seed-value             Seed value for the random number generator. (default: None)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="rm-dup-tss">
<h2>rm_dup_tss<a class="headerlink" href="#rm-dup-tss" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> If several transcripts of a gene share the same tss, select only one.</p>
<p><strong>Example:</strong> Use rm_dup_tss to select transcripts that will be used for mk_matrix -k 5 (see later).</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk rm_dup_tss -i simple.gtf | gtftk select_by_key -k feature -v transcript
chr1	gtftk	transcript	125	138	.	+	.	gene_id &quot;G0001&quot;; transcript_id &quot;G0001T001&quot;;
chr1	gtftk	transcript	180	189	.	+	.	gene_id &quot;G0002&quot;; transcript_id &quot;G0002T001&quot;;
chr1	gtftk	transcript	50	61	.	-	.	gene_id &quot;G0003&quot;; transcript_id &quot;G0003T001&quot;;
chr1	gtftk	transcript	65	76	.	+	.	gene_id &quot;G0004&quot;; transcript_id &quot;G0004T001&quot;;
chr1	gtftk	transcript	33	47	.	-	.	gene_id &quot;G0005&quot;; transcript_id &quot;G0005T001&quot;;
chr1	gtftk	transcript	22	35	.	-	.	gene_id &quot;G0006&quot;; transcript_id &quot;G0006T001&quot;;
chr1	gtftk	transcript	107	116	.	+	.	gene_id &quot;G0007&quot;; transcript_id &quot;G0007T001&quot;;
chr1	gtftk	transcript	210	222	.	-	.	gene_id &quot;G0008&quot;; transcript_id &quot;G0008T001&quot;;
chr1	gtftk	transcript	3	14	.	-	.	gene_id &quot;G0009&quot;; transcript_id &quot;G0009T001&quot;;
chr1	gtftk	transcript	176	186	.	+	.	gene_id &quot;G0010&quot;; transcript_id &quot;G0010T001&quot;;
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk rm_dup_tss -h
  Usage: gtftk rm_dup_tss [-i GTF] [-o GTF] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     If several transcripts of a gene share the same TSS, select one transcript per TSS.

  Notes:
     *  The alphanumeric order of transcript_id is used to select the representative of a TSS.

Argument:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-by-go">
<h2>select_by_go<a class="headerlink" href="#select-by-go" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select genes from a GTF file using a Gene Ontology ID (e.g <a class="reference external" href="GO:0050789">GO:0050789</a>).</p>
<p><strong>Example:</strong> Select genes with transcription factor activity from the GTF. They could be used subsequently to test their epigenetic features (see later).</p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ # gtftk select_by_go -s hsapiens -i mini_real.gtf.gz | gtftk select_by_key -k feature -v gene | gtftk tabulate -k gene_id,gene_name -Hun | head -6
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_go -h
  Usage: gtftk select_by_go [-i GTF] [-o GTF] [-g go_id] (-l | -s species) [-n] [-p1 http_proxy] [-p2 https_proxy] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select lines/genes from a GTF file using a Gene Ontology ID (e.g GO:0097194).

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -g, --go-id                  The GO ID (with or without &quot;GO:&quot; prefix). (default: GO:0003700)
 -l, --list-datasets          Do not select lines. Only get a list of available datasets/species. (default: False)
 -s, --species                The dataset/species. (default: None)
 -n, --invert-match           Not/invert match. (default: False)
 -p1, --http-proxy            Use this http proxy (not tested/experimental). (default: )
 -p2, --https-proxy           Use this https proxy (not tested/experimental). (default: )

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-by-tx-size">
<h2>select_by_tx_size<a class="headerlink" href="#select-by-tx-size" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select transcript based on their size (i.e size of mature/spliced transcript).</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk feature_size -t mature_rna -i simple.gtf |  gtftk select_by_tx_size -m 14 | gtftk tabulate -n -k gene_id,transcript_id,feat_size
gene_id	transcript_id	feat_size
G0001	G0001T002	14
G0001	G0001T001	14
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_by_tx_size -h
  Usage: gtftk select_by_tx_size [-i GTF] [-o GTF] [-m min_size] [-M max_size] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select transcript based on their size (i.e size of mature/spliced transcript).

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -m, --min-size               Minimum size. (default: 0)
 -M, --max-size               Maximum size. (default: 1000000000)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="select-most-5p-tx">
<h2>select_most_5p_tx<a class="headerlink" href="#select-most-5p-tx" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Select the most 5’ transcript of each gene.</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_most_5p_tx -i simple.gtf | gtftk select_by_key -k feature -v transcript| gtftk tabulate -k gene_id,transcript_id
gene_id	transcript_id
G0001	G0001T002
G0002	G0002T001
G0003	G0003T001
G0004	G0004T002
G0005	G0005T001
G0006	G0006T001
G0007	G0007T001
G0008	G0008T001
G0009	G0009T002
G0010	G0010T001
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk select_most_5p_tx -h
  Usage: gtftk select_most_5p_tx [-i GTF] [-o GTF] [-g] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select the most 5&#39; transcript of each gene.

  Notes:
     *  If several transcript share the samemost 5&#39; TSS, only one transcript is selected.

optional arguments:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -g, --keep-gene-lines        Add gene lines to the output (default: False)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
<hr class="docutils" />
<section id="short-long">
<h2>short_long<a class="headerlink" href="#short-long" title="Permalink to this headline">¶</a></h2>
<p><strong>Description:</strong> Get the shortest or longest transcript of each gene</p>
<p><strong>Example:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk short_long -i simple.gtf | gtftk select_by_key -k feature -v transcript| gtftk tabulate -k gene_id,transcript_id
gene_id	transcript_id
G0001	G0001T002
G0002	G0002T001
G0003	G0003T001
G0004	G0004T002
G0005	G0005T001
G0006	G0006T002
G0007	G0007T001
G0008	G0008T001
G0009	G0009T002
G0010	G0010T001
</pre></div>
</div>
<p><strong>Arguments:</strong></p>
<div class="highlight-text notranslate"><div class="highlight"><pre><span></span>$ gtftk short_long -h
  Usage: gtftk short_long [-i GTF] [-o GTF] [-l] [-g] [-h] [-V [verbosity]] [-D] [-C] [-K tmp_dir] [-A] [-L logger_file] [-W write_message_to_file]

  Description: 

     Select the shortest mature transcript (i.e without introns) for each gene or the longest if the -l
     arguments is used.

  Notes:
     *

Argument:
 -i, --inputfile              Path to the GTF file. Default to STDIN (default: &lt;stdin&gt;)
 -o, --outputfile             Output file. (default: &lt;stdout&gt;)
 -l, --longs                  Take the longest transcript of each gene (default: False)
 -g, --keep-gene-lines        Add gene lines to the output (default: False)

Command-wise optional arguments:
 -h, --help                   Show this help message and exit.
 -V, --verbosity              Set output verbosity ([0-3]). (default: 0)
 -D, --no-date                Do not add date to output file names. (default: False)
 -C, --add-chr                Add &#39;chr&#39; to chromosome names before printing output. (default: False)
 -K, --tmp-dir                Keep all temporary files into this folder. (default: None)
 -A, --keep-all               Try to keep all temporary files even if process does not terminate normally. (default: False)
 -L, --logger-file            Stores the arguments passed to the command into a file. (default: None)
 -W, --write-message-to-file  Store all message into a file. (default: None)
</pre></div>
</div>
</section>
</section>


            <div class="clearer"></div>
          </div>
        </div>
      </div>
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
  <div>
    <h3><a href="index.html">Table of Contents</a></h3>
    <ul>
<li><a class="reference internal" href="#">Commands from section ‘selection’</a><ul>
<li><a class="reference internal" href="#select-by-key">select_by_key</a></li>
<li><a class="reference internal" href="#select-by-regexp">select_by_regexp</a></li>
<li><a class="reference internal" href="#select-by-intron-size">select_by_intron_size</a></li>
<li><a class="reference internal" href="#select-by-max-exon-nb">select_by_max_exon_nb</a></li>
<li><a class="reference internal" href="#select-by-loc">select_by_loc</a></li>
<li><a class="reference internal" href="#select-by-nb-exon">select_by_nb_exon</a></li>
<li><a class="reference internal" href="#select-by-numeric-value">select_by_numeric_value</a></li>
<li><a class="reference internal" href="#random-list">random_list</a></li>
<li><a class="reference internal" href="#random-tx">random_tx</a></li>
<li><a class="reference internal" href="#rm-dup-tss">rm_dup_tss</a></li>
<li><a class="reference internal" href="#select-by-go">select_by_go</a></li>
<li><a class="reference internal" href="#select-by-tx-size">select_by_tx_size</a></li>
<li><a class="reference internal" href="#select-most-5p-tx">select_most_5p_tx</a></li>
<li><a class="reference internal" href="#short-long">short_long</a></li>
</ul>
</li>
</ul>

  </div>
  <div>
    <h4>Previous topic</h4>
    <p class="topless"><a href="editing.html"
                          title="previous chapter">Commands from section ‘Editing’</a></p>
  </div>
  <div>
    <h4>Next topic</h4>
    <p class="topless"><a href="conversion.html"
                          title="next chapter">Commands from section ‘conversion’</a></p>
  </div>
  <div role="note" aria-label="source link">
    <h3>This Page</h3>
    <ul class="this-page-menu">
      <li><a href="_sources/selection.rst.txt"
            rel="nofollow">Show Source</a></li>
    </ul>
   </div>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" autocomplete="off" autocorrect="off" autocapitalize="off" spellcheck="false"/>
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="clearer"></div>
    </div>
    <div class="related" role="navigation" aria-label="related navigation">
      <h3>Navigation</h3>
      <ul>
        <li class="right" style="margin-right: 10px">
          <a href="genindex.html" title="General Index"
             >index</a></li>
        <li class="right" >
          <a href="py-modindex.html" title="Python Module Index"
             >modules</a> |</li>
        <li class="right" >
          <a href="conversion.html" title="Commands from section ‘conversion’"
             >next</a> |</li>
        <li class="right" >
          <a href="editing.html" title="Commands from section ‘Editing’"
             >previous</a> |</li>
        <li class="nav-item nav-item-0"><a href="index.html">gtftk 1.6.2 documentation</a> &#187;</li>
        <li class="nav-item nav-item-this"><a href="">Commands from section ‘selection’</a></li> 
      </ul>
    </div>
    <div class="footer" role="contentinfo">
        &#169; Copyright 2018, F. Lopez and D. Puthier.
      Last updated on Mar 24, 2022.
      Created using <a href="https://www.sphinx-doc.org/">Sphinx</a> 4.4.0.
    </div>
  </body>
</html>